projects/ngrx-auto-entity/src/lib/actions/util.ts

```diff
@@ -32,3 +33,9 @@ export const setType = (actionType: string, info: IEntityInfo): string => {
 
   return actionType.replace('Entity', entity);
 };
+
+export const setActionType = <TModel, T extends string>(actionType: string, type: TNew<TModel>): T => {
```

Comment:
The one "sketchy" part of the update here. The generic `T` here extends `string`. This may potentially limit the strength of the typing, as technically T is supposed to be unique per action. Given the dynamic nature of this library, even if I was to have made extensive breaking changes to the library to eliminate our legacy actions and create a set of brand new NgRx 8 style actions...I am still fairly certain that we would have still had to drop to the more general `string` for action types at some point. I just hope that NgRx remains backwards compatible in this respect.